src/store/highlight/actions.ts

```diff
@@ -13,6 +13,57 @@ type Payload = {
     payload: SelectionItem | null;
 };
 
+declare global {
+    interface Document {
+        createNodeIterator(
+            root: Node,
+            whatToShow?: number,
+            filter?: NodeFilter | Function,
+            entityReferenceExpansion?: boolean,
+        ): NodeIterator;
+    }
+}
+
+export const getRects = (range: Range): DOMRect[] => {
+    const iterator = document.createNodeIterator(
+        range.commonAncestorContainer,
+        NodeFilter.SHOW_TEXT, // only show Text nodes
+        function acceptNode() {
+            return NodeFilter.FILTER_ACCEPT;
+        },
+        false,
+    );
+
+    const newRange = document.createRange();
+    const rects = [];
+    let currentNode = iterator.nextNode();
+    while (currentNode) {
+        if (rects.length === 0 && currentNode !== range.startContainer) {
+            currentNode = iterator.nextNode();
+            // eslint-disable-next-line no-continue
+            continue;
+        }
+
+        newRange.selectNodeContents(currentNode);
+        if (currentNode === range.startContainer) {
+            newRange.setStart(currentNode, range.startOffset);
+        }
+        if (currentNode === range.endContainer) {
+            newRange.setEnd(currentNode, range.endOffset);
+        }
+        rects.push(newRange.getBoundingClientRect());
+
+        if (currentNode === range.endContainer) {
+            break;
+        }
+
+        currentNode = iterator.nextNode();
+    }
+    newRange.detach();
```

Comment:
Seems like this is not needed anymore? https://developer.mozilla.org/en-US/docs/Web/API/Range/detach